src/plugins/flexibleLayout/components/frame.vue

```diff
@@ -21,104 +21,84 @@
-import DropHint from './dropHint.vue';
 import ResizeHandle from './resizeHandle.vue';
-import FrameHeader from '../../../ui/components/utils/frameHeader.vue';
+import SubobjectView from '../../displayLayout/components/SubobjectView.vue';
+import isEditingMixin from '../mixins/isEditing';
 
 export default {
     inject: ['openmct', 'domainObject'],
-    props: ['frame', 'index', 'containerIndex', 'isEditing', 'isDragging'],
+    props: ['frame', 'index', 'containerIndex'],
+    mixins: [isEditingMixin],
     data() {
         return {
+            item: {domainObject: {}},
             noFrame: this.frame.noFrame
         }
     },
     components: {
-        ObjectView,
-        DropHint,
         ResizeHandle,
-        FrameHeader
+        SubobjectView
     },
     methods: {
-        initDrag(event) {
-            this.$emit('frame-drag-from', this.index);
-        },
-        dropHandler(event) {
-            this.$emit('frame-drop-to', this.index, event);
+        setDomainObject(object) {
+            this.item.domainObject = object;
+            this.setSelection();
         },
-        continueDrag(event) {
-            if (!this.isDragging) {
-                this.isDragging = true;
-            }
+        setSelection() {
+           
+            let context = {
+                item: this.item.domainObject,
+                addContainer: this.addContainer,
+                type: 'frame',
+                index: this.index
+            };
+
+            this.unsubscribeSelection = this.openmct.selection.selectable(this.$refs.frame, context, false);
+            
+            this.openmct.objects.observe(this.domainObject, `configuration.containers[${this.containerIndex}].frames[${this.index}].noFrame`, this.toggleFrame);
+
         },
-        deleteFrame() {
-            this.$emit('delete-frame', this.index);
+        initDrag(event) {
+            this.$emit('frame-drag-from', this.index);
+            event.dataTransfer.setData('frameid', JSON.stringify([this.frame.id, this.containerIndex]));
         },
-        addContainer() {
-            this.$emit('add-container');
+        endDrag(event) {
+            event.dataTransfer.clearData('frameid');
         },
         toggleFrame(v) {
             this.noFrame = v;
         }
     },
     mounted() {
-
-        if (this.frame.domainObject.identifier) {
-                let context = {
-                item: this.frame.domainObject,
-                method: this.deleteFrame,
-                addContainer: this.addContainer,
-                type: 'frame',
-                index: this.index
-            }
-
-            this.unsubscribeSelection = this.openmct.selection.selectable(this.$refs.frame, context, false);
-            
-            this.openmct.objects.observe(this.domainObject, `configuration.containers[${this.containerIndex}].frames[${this.index}].noFrame`, this.toggleFrame);
+        if (this.frame.domainObjectIdentifier.key) {
```

Comment:
It's possible that `key` could have a falsy value, so it would be better to check for `this.frame.domainObjectIdentifier` instead of the `key` property.